__tests__/endpoints.test.js

```diff
@@ -161,3 +161,26 @@ describe('GET /api/articles/:article_id/comments', () => {
         })
     })
 })
+describe('POST /api/articles/:article_id/comments', () => {
+    test.only('responds with code 201 and the message object', () => {
+        const testComment = {
+            "user": "lurker",
+            "body": "Hi there this is a test body"
+        }
+        return request(app)
+        .post('/api/articles/9/comments')
+        .send(testComment)
+        .expect(201)
+        .then((res) => {
+            const comment = res.body.comment;
+            expect(comment).toMatchObject({
+                comment_id: expect.any(Number),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                article_id: expect.any(Number)
+            });
+        })
+    })
```

Comment:
Great work on the main functionality bit here - but this needs a bit of consideration for error handling. There's quite a lot of possible failure points in this request, in the article_id and the request body. Try implementing the following:

    - Status 201, ignores unnecessary properties
    - Status 400, invalid article ID, e.g. string of "not-an-id"
    - Status 404, non existent article ID, e.g. 0 or 9999
    - Status 400, missing required field(s), e.g. no username or body properties
    - Status 404, username does not exist ( this refers to that foreign key constraint to the users table we mentioned)